server/resp/src/main/java/org/infinispan/server/resp/Resp3Handler.java

```diff
@@ -53,180 +52,55 @@ public RespRequestHandler handleRequest(ChannelHandlerContext ctx, String type,
             ctx.writeAndFlush(bufferToWrite);
             break;
          case "SET":
-            performSet(ctx, cache, arguments.get(0), arguments.get(1), -1, type, statusOK());
-            break;
+            return performSet(ctx, cache, arguments.get(0), arguments.get(1), -1, type, statusOK());
          case "GET":
             byte[] keyBytes = arguments.get(0);
 
-            cache.getAsync(keyBytes)
-                  .whenComplete((innerValueBytes, t) -> {
-                     if (t != null) {
-                        log.trace("Exception encountered while performing GET", t);
-                        handleThrowable(ctx, t);
-                     } else if (innerValueBytes != null) {
-                        int length = innerValueBytes.length;
-                        ByteBuf buf = RespRequestHandler.stringToByteBufWithExtra("$" + length + "\r\n", ctx.alloc(), length + 2);
-                        buf.writeBytes(innerValueBytes);
-                        buf.writeByte('\r').writeByte('\n');
-                        ctx.writeAndFlush(buf);
-                     } else {
-                        ctx.writeAndFlush(RespRequestHandler.stringToByteBuf("$-1\r\n", ctx.alloc()));
-                     }
-                  });
-            break;
-         case "DEL":
-            int keysToRemove = arguments.size();
-            if (keysToRemove == 1) {
-               keyBytes = arguments.get(0);
-               cache.removeAsync(keyBytes)
-                     .whenComplete((prev, t) -> {
-                        if (t != null) {
-                           log.trace("Exception encountered while performing DEL", t);
-                           handleThrowable(ctx, t);
-                           return;
-                        }
-                        ctx.writeAndFlush(RespRequestHandler.stringToByteBuf(":" + (prev == null ? "0" : "1") +
-                              "\r\n", ctx.alloc()));
-                     });
-            } else if (keysToRemove == 0) {
-               // TODO: is this an error?
-               ctx.writeAndFlush(RespRequestHandler.stringToByteBuf(":0\r\n", ctx.alloc()));
-            } else {
-               AtomicInteger removes = new AtomicInteger();
-               AggregateCompletionStage<AtomicInteger> deleteStages = CompletionStages.aggregateCompletionStage(removes);
-               for (byte[] keyBytesLoop : arguments) {
-                  deleteStages.dependsOn(cache.removeAsync(keyBytesLoop)
-                        .thenAccept(prev -> {
-                           if (prev != null) {
-                              removes.incrementAndGet();
-                           }
-                        }));
+            return stageToReturn(cache.getAsync(keyBytes), ctx, (innerValueBytes, t) -> {
+               if (t != null) {
+                  log.trace("Exception encountered while performing GET", t);
```

Comment:
nitpick, I would probably check `log.isTraceEnabled()` in all places where `log.trace()` is used.

Reply:
`really necessary` -> does it mean some intensive computation?

Reply:
If it requires additional allocations or computation to generate the message.